File: assets/codebase/menu/MenuItemRuntime.cs
namespace city.menu {
    /// <summary>
    /// Stores the live scene references associated with one baked city menu item.
    /// </summary>
    internal sealed class MenuItemRuntime {
        /// <summary>
        /// Initializes one baked city menu item runtime record.
        /// </summary>
        /// <param name="definition">Serialized item metadata component.</param>
        /// <param name="index">Enabled-item index inside the owning panel.</param>
        /// <param name="entity">Owning row entity.</param>
        /// <param name="background">Rounded rectangle used for selection visuals.</param>
        public MenuItemRuntime(
            MenuItemComponent definition,
            int index,
            Entity entity,
            RoundedRectComponent background) {
            Definition = definition ?? throw new ArgumentNullException(nameof(definition));
            Index = index;
            OwnerEntity = entity ?? throw new ArgumentNullException(nameof(entity));
            Background = background ?? throw new ArgumentNullException(nameof(background));
        }

        /// <summary>
        /// Gets the serialized metadata component for the item.
        /// </summary>
        public MenuItemComponent Definition { get; }

        /// <summary>
        /// Gets the enabled-item index inside the owning panel.
        /// </summary>
        public int Index { get; }

        /// <summary>
        /// Gets the entity that owns the baked row visuals.
        /// </summary>
        public Entity OwnerEntity { get; }

        /// <summary>
        /// Gets the rounded rectangle used for selection-state color updates.
        /// </summary>
        public RoundedRectComponent Background { get; }
    }
}

File: windows-build-20260602-colored-cube-grid-direct-fix1/cooked/code/gameplay/MenuItemRuntime.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "MenuItemRuntime.hpp"
#include "MenuItemComponent.hpp"
#include "runtime/native_exceptions.hpp"
#include "MenuItemRuntime.hpp"
#include "runtime/native_exceptions.hpp"

RoundedRectComponent* MenuItemRuntime::get_Background()
{
return this->Background;
}

::MenuItemComponent* MenuItemRuntime::get_Definition()
{
return this->Definition;
}

int32_t MenuItemRuntime::get_Index()
{
return this->Index;
}

Entity* MenuItemRuntime::get_OwnerEntity()
{
return this->OwnerEntity;
}

MenuItemRuntime::MenuItemRuntime(::MenuItemComponent* definition, int32_t index, Entity* entity, RoundedRectComponent* background) : Background(), Definition(), Index(0), OwnerEntity()
{
this->Definition = (definition != nullptr ? definition : throw new ArgumentNullException("definition"));
this->Index = index;
this->OwnerEntity = (entity != nullptr ? entity : throw new ArgumentNullException("entity"));
this->Background = (background != nullptr ? background : throw new ArgumentNullException("background"));
}

